Formula/texmaker.rb

```diff
@@ -0,0 +1,61 @@
+class Texmaker < Formula
+  desc "Free cross-platform LaTeX editor since 2003"
+  homepage "https://www.xm1math.net/texmaker/"
+  url "https://www.xm1math.net/texmaker/texmaker-5.1.2.tar.bz2"
+  sha256 "526896f2c1ae561130eec7aae815b9dcda9e8eeb772b6541d0dc94ce91a71044"
+  license "GPL-2.0-only"
+
+  depends_on xcode: :build
+  depends_on "qt"
+
+  on_linux do
+    depends_on "gcc"
+  end
+
+  fails_with gcc: "5"
+
+  patch :p0, :DATA
+
+  def install
+    args = [
+      "PREFIX=#{prefix}",
+      "DESKTOPDIR=#{share}/applications",
+      "ICONDIR=#{share}/pixmaps",
+      "METAINFODIR=#{share}/metainfo",
+      "QMAKE_CXXFLAGS=#{ENV.cxxflags}",
+      "-config", "release",
+      "-spec"
+    ]
+    os = OS.mac? ? "macx" : OS.kernel_name.downcase
+    compiler = ENV.compiler.to_s.start_with?("gcc") ? "g++" : ENV.compiler
+    args << "#{os}-#{compiler}"
+
+    system Formula["qt"].opt_bin/"qmake", *args
+    system "make"
+    system "make", "install"
+  end
+
+  test do
+    assert_predicate bin/"texmaker", :exist?, "texmaker must exist"
+    assert_predicate bin/"texmaker", :executable?, "texmaker must be executable"
```

Comment:
Can you propose a more substantial test than checking for the existence of certain files, or checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

Reply:
Unfortunately the binary does not allow for sth like `--help` or `--version`. It will always start the GUI.

Reply:
I could write another patch which patches the `main()` function to add sth like `--version`. But then again, this is probably nicer to have upstream.

Reply:
Can we try to run the binary briefly and check that it doesn't crash? Does it generate any logs?

Reply:
I think it would need an X server to run. Is this ok for such a test? I don't think it creates any logs.

Reply:
Does that mean users who install this formula need to install an X server separately to use this?

Reply:
I don't know. Whatever Qt requires. I think Qt has many possible backends. I'm not sure which backends are installed by Homebrew.